drivers/vtencent/util.go

```diff
@@ -0,0 +1,392 @@
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/alist-org/alist/v3/drivers/base"
+	"github.com/alist-org/alist/v3/internal/driver"
+	"github.com/alist-org/alist/v3/internal/model"
+	"github.com/alist-org/alist/v3/pkg/utils"
+	"github.com/go-resty/resty/v2"
+	"github.com/google/uuid"
+)
+
+func (d *Vtencent) request(url, method string, callback base.ReqCallback, resp interface{}) ([]byte, error) {
+	req := base.RestyClient.R()
+	req.SetHeaders(map[string]string{
+		"cookie":       d.Cookie,
+		"content-type": "application/json",
+		"origin":       d.conf.origin,
+		"referer":      d.conf.referer,
+	})
+	if callback != nil {
+		callback(req)
+	} else {
+		req.SetBody("{}")
+	}
+	if resp != nil {
+		req.SetResult(resp)
+	}
+	res, err := req.Execute(method, url)
+	if err != nil {
+		return nil, err
+	}
+	code := utils.Json.Get(res.Body(), "Code").ToString()
+	if code != "Success" {
+		switch code {
+		case "AuthFailure.SessionInvalid":
+			if err != nil {
+				return nil, errors.New(code)
+			}
+		default:
+			return nil, errors.New(code)
+		}
+		return d.request(url, method, callback, resp)
+	}
+	return res.Body(), nil
+}
+
+func (d *Vtencent) ugcRequest(url, method string, callback base.ReqCallback, resp interface{}) ([]byte, error) {
+	req := base.RestyClient.R()
+	req.SetHeaders(map[string]string{
+		"cookie":       d.Cookie,
+		"content-type": "application/json",
+		"origin":       d.conf.origin,
+		"referer":      d.conf.referer,
+	})
+	if callback != nil {
+		callback(req)
+	} else {
+		req.SetBody("{}")
+	}
+	if resp != nil {
+		req.SetResult(resp)
+	}
+	res, err := req.Execute(method, url)
+	if err != nil {
+		return nil, err
+	}
+	code := utils.Json.Get(res.Body(), "Code").ToInt()
+	if code != 0 {
+		message := utils.Json.Get(res.Body(), "message").ToString()
+		if len(message) == 0 {
+			message = utils.Json.Get(res.Body(), "msg").ToString()
+		}
+		return nil, errors.New(message)
+	}
+	return res.Body(), nil
+}
+
+func (d *Vtencent) LoadUser() (string, error) {
+	api := "https://api.vs.tencent.com/SaaS/Account/DescribeAccount"
+	res, err := d.request(api, http.MethodPost, func(req *resty.Request) {}, nil)
+	if err != nil {
+		return "", err
+	}
+	return utils.Json.Get(res, "Data", "TfUid").ToString(), nil
+}
+
+func (d *Vtencent) GetFiles(dirId string) ([]File, error) {
+	api := "https://api.vs.tencent.com/PaaS/Material/SearchResource"
+	form := fmt.Sprintf(`{
+		"Text":"",
+		"Text":"",
+		"Offset":0,
+		"Limit":20000,
+		"Sort":{"Field":"%s","Order":"%s"},
+		"CreateTimeRanges":[],
+		"MaterialTypes":[],
+		"ReviewStatuses":[],
+		"Tags":[],
+		"SearchScopes":[{"Owner":{"Type":"PERSON","Id":"%s"},"ClassId":%s,"SearchOneDepth":true}]
+	}`, d.Addition.OrderBy, d.Addition.OrderDirection, d.TfUid, dirId)
+	var dat map[string]interface{}
+	if err := json.Unmarshal([]byte(form), &dat); err != nil {
+		return []File{}, err
+	}
+	var resps RspFiles
+	rsp, err := d.request(api, http.MethodPost, func(req *resty.Request) {
+		req.SetBody(dat)
```

Comment:
```suggestion
		req.SetBody(form)
```
设置"content-type": "application/json"头后，可以直接发json字符串。